nntrainer/tensor/cache_pool.cpp

```diff
@@ -106,23 +136,41 @@ void CachePool::invalidate(unsigned int id) {
   }
 }
 
+unsigned int CachePool::requestMemory(size_t bytes, unsigned int start_time,
+                                      unsigned int end_time,
+                                      std::vector<unsigned int> exec_order,
+                                      TensorLifespan lifespan) {
+  auto id = MemoryPool::requestMemory(bytes, start_time, end_time, exec_order,
+                                      lifespan);
+
+  const CachePolicy policy = convertTensorLifespanToCachePolicy(lifespan);
+
+  policies.push_back(policy);
```

Comment:
I'm not sure whether the id is not considered here.

Reply:
It's for simplying the implementaion. We already knows that the `id` is size of each information's length(validity, exec_order).
To check the consititency, it checks the `id` with policies length (line 155).